storm-core/src/jvm/org/apache/storm/cluster/StormClusterStateImpl.java

```diff
@@ -219,6 +219,8 @@ public void stateChanged(CuratorFramework curatorFramework, ConnectionState conn
                 LOG.info("Connection state listener invoked, zookeeper connection state has changed to {}", connectionState);
                 if (connectionState.equals(ConnectionState.RECONNECTED)) {
                     LOG.info("Connection state has changed to reconnected so setting nimbuses entry one more time");
+                    // explicit delete for ephmeral node to ensure this session creates the entry.
+                    stateStorage.delete_node(ClusterUtils.nimbusPath(nimbusId));
```

Comment:
@harshach 
This is for edge case for ephemeral node not deleted while session is closed.
Please refer http://issues.apache.org/jira/browse/STORM-1941.

I know it's odd but it is happening, and other article states this behavior.
https://www.box.com/blog/a-gotcha-when-using-zookeeper-ephemeral-nodes/